resources/benchmark-runner.mjs

```diff
@@ -31,28 +31,94 @@ class Page {
 
     querySelector(selector)
     {
-        return this._frame.contentDocument.querySelector(selector);
+        const element = this._frame.contentDocument.querySelector(selector);
+        if (element === null) return null
+        return this._wrapElement(element);
     }
 
     querySelectorAll(selector)
     {
-        return this._frame.contentDocument.querySelectorAll(selector);
+        const elements = Array.from(this._frame.contentDocument.querySelectorAll(selector));
+        for (let i = 0; i < elements.length; i++)
+            elements[i] = this._wrapElement(elements[i]);
+        return elements
     }
 
     getElementById(id)
     {
-        return this._frame.contentDocument.getElementById(id);
+        const element = this._frame.contentDocument.getElementById(id);
+        if (element === null) return null
+        return this._wrapElement(element)
     }
 
-    triggerEnter(element, type)
+    eval(source)
+    {
+        this._frame.contentWindow.eval(source);
+        return null;
+    }
+
+    _wrapElement(element)
+    {
+        return new PageElement(element);
+    }
+}
+
+const NATIVE_OPTIONS = {
+    bubbles: true,
+    cancellable: true,
+}
+
+class PageElement {
+    constructor(node)
+    {
+        this._node = node
+    }
+
+    setValue(value)
+    {
+        this._node.value = value
+    }
+
+    click()
+    {
+        this._node.click()
+    }
+
+    focus()
+    {
+        this._node.focus()
+    }
+
+    dispatchEvent(eventName, options = NATIVE_OPTIONS)
+    {
+        if (eventName === 'submit')
+            // FIXME FireFox doesn't like `new Event('submit')
```

Comment:
Again, this comment seems outdated at this point.
new Event('submit') works in Firefox.